massa-execution-worker/src/context.rs

```diff
@@ -635,6 +635,54 @@ impl ExecutionContext {
         )
     }
 
+    /// Try to slash 1 roll from the denounced address.
+    ///
+    /// # Arguments
+    /// * `denounced_addr`: address to slash the roll or deferred credit from
+    pub fn try_slash_roll(
+        &mut self,
+        denounced_addr: &Address,
+        roll_count: u64,
+    ) -> Result<Amount, ExecutionError>
+    {
+        let slashed_coins;
+
+        // First try to slash roll if any
+        if self.speculative_roll_state.try_burn_rolls(
+            denounced_addr,
+            roll_count,
+        ).is_err() {
+
+            // if no roll were slashed, that can likely be done from deferred credit
+            let credits = self.speculative_roll_state.get_address_deferred_credits(denounced_addr, self.slot);
```

Comment:
This function returns all deferred credits after a given slot, included.

But deferred credits are credited at the beginning of the slot execution (before the block is executed) by calling context::execute_deferred_credits.
This means that if there is a deferred credit for the current slot, we cannot slash it because it was already credited.

In summary: credits targeting exactly "self.slot" should not be taken into account

Reply:
done